include/swift/AST/DiagnosticsFrontend.def

```diff
@@ -444,11 +444,19 @@ NOTE(compiled_module_ignored_reason,none,
      "|it belongs to a framework in the SDK"
      "|loading from module interfaces is preferred"
      "|it's a compiler host module"
```

Comment:
I believe it's for macros and refers to a cross-compilation concern, or at least bootstrapping. 

@bnbarham Could you remind us why we ignore swiftmodules when `isInResourceHostDir`?

Reply:
This is really just an issue in our build/test infrastructure, as the modules won't even be there in the toolchain (or at least, they shouldn't be if they are 😅). The issue is that these libraries are built by the "host" (builder/<whatever other name people agree on>) compiler, but are then read by the just-built compiler when eg. building libraries/executables in the tests. An alternative to this would be building the `swiftmodule` itself elsewhere during the build, I'd be open to that rather than this hack if you'd prefer as well.

Reply:
We'll need an option in build systems to not install swiftmodules at some point. For this PR we could improve the proposed diagnostic, it sounds like it will be mostly us seeing it.